src/index.ts

```
@@ -13,8 +13,33 @@ import { loggingMiddleware } from './middlewares/logger'
 import { makeSchema } from './schema'
 import { factory } from './utils/log'
 
+import * as Sentry from '@sentry/node'
+Sentry.init({
+  dsn: config.SENTRY_DSN,
+  enabled: Boolean(config.SENTRY_DSN),
+  environment: process.env.NODE_ENV,
+})
+
 const logger = factory.getLogger('index')
 
+const handleError = (error: GraphQLError): void => {
+  logger.error(
+    'Uncaught error in GraphQL. Original error: ',
+    error.originalError,
+  )
+  Sentry.captureException(error)
```

would it be possible to do some more lovin with scoping? I think this'll kinda catch all errors and breadcrumbs from the beginning of time

---

I'm not sure that I follow 🙁 I'd love to fix this however!

---

I'm thinking something like this https://github.com/HedvigInsurance/web-onboarding/blob/master/src/server/middleware/enhancers.ts#L88-L100

---

From what I can tell Apollo Server eats the error before it comes back to Koa, making a regular Koa Middleware not the ideal approach (?). Nor does Apollo Server offer any nice places to perform the 'wrapping' we're looking for in an around-style middleware.

Apollo is starting to make me a little bit sad 🙁

---

New implementation - Ping! @palmfjord